# DSCI 100 Project Proposal 

In [2]:
import pandas as pd 
import altair as alt

# (1) Data Description:

**players.csv Data**

In [3]:
players_data = pd.read_csv("https://drive.google.com/uc?export=download&id=1Mw9vW0hjTJwRWx0bDXiSpYsO3gKogaPz") 
players_data 

experience  subscribe                                        hashedEmail  \
0          Pro       True  f6daba428a5e19a3d47574858c13550499be23603422e6...   
1      Veteran       True  f3c813577c458ba0dfef80996f8f32c93b6e8af1fa9397...   
2      Veteran      False  b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3...   
3      Amateur       True  23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4f...   
4      Regular       True  7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb...   
..         ...        ...                                                ...   
191    Amateur       True  b6e9e593b9ec51c5e335457341c324c34a2239531e1890...   
192    Veteran      False  71453e425f07d10da4fa2b349c83e73ccdf0fb3312f778...   
193    Amateur      False  d572f391d452b76ea2d7e5e53a3d38bfd7499c7399db29...   
194    Amateur      False  f19e136ddde68f365afc860c725ccff54307dedd13968e...   
195        Pro       True  d9473710057f7d42f36570f0be83817a4eea614029ff90...   

     played_hours       name             gender  age  individualId  \
0            30.3     Morgan               Male    9           NaN   
1             3.8  Christian               Male   17           NaN   
2             0.0      Blake               Male   17           NaN   
3             0.7      Flora             Female   21           NaN   
4             0.1      Kylie               Male   21           NaN   
..            ...        ...                ...  ...           ...   
191           0.0     Bailey             Female   17           NaN   
192           0.3     Pascal               Male   22           NaN   
193           0.0      Dylan  Prefer not to say   17           NaN   
194           2.3     Harlow               Male   17           NaN   
195           0.2      Ahmed              Other   91           NaN   

     organizationName  
0                 NaN  
1                 NaN  
2                 NaN  
3                 NaN  
4                 NaN  
..                ...  
191               NaN  
192               NaN  
193               NaN  
194               NaN  
195               NaN  

[196 rows x 9 columns]

**Data Description of players_data**

There are 196 individuals recorded in this dataset and there are 9 variables.

In [4]:
data = {
    "Variable Name": [
        "Experience", "Subscribe", "hashedEmail", "played_hours",
        "name", "gender", "age", "individualid", "organizationName"
    ],
    "Data Type": [
        "Categorical", "Boolean", "Text/String", "Numeric (Float)",
        "Text/String", "Categorical", "Numeric (Integer)", "Text/String", "Text/String"
    ],
    "Description": [
        "User’s experience level/skill",
        "Subscription Status",
        "Anonymized email identifier",
        "Total Hours Played",
        "User’s first name",
        "User’s Gender identity",
        "User's Age in years",
        "User’s ID",
        "Associated Organization"
    ]
}

df = pd.DataFrame(data)
df

Variable Name          Data Type                    Description
0        Experience        Categorical  User’s experience level/skill
1         Subscribe            Boolean            Subscription Status
2       hashedEmail        Text/String    Anonymized email identifier
3      played_hours    Numeric (Float)             Total Hours Played
4              name        Text/String              User’s first name
5            gender        Categorical         User’s Gender identity
6               age  Numeric (Integer)            User's Age in years
7      individualid        Text/String                      User’s ID
8  organizationName        Text/String        Associated Organization

### Summary Statistics

**Numerical Variables:**
- **played_hours**: Ranges from 0 to 240.0 hours. The distribution is highly right-skewed, with majority of players having played fewer than 30 hours.
- **age**: Ranges from 9 to 91 years. The majority of players fall within the 15-25 age range.

**Categorical Variables:**
- **experience**: Amateur (63 players), Beginner (35 players), Pro (14 players), Regular (36 players), Veteran (48 players)
- **gender**: Male (161 players), Female (37 players), Other (1 players), Prefer not to say (11 players), Non-Binary (15 players), Agender (2 players), Two spirited (6 players).
- **subscribe**: True (144 players), False (52 players)

**Data Quality Issues Quantified:**
- 2 columns (individualId, organizationName) are completely empty (100% NaN values)
- Approximately 83 players have 0 hours played, which may indicate they registered but never engaged
- 2 age outliers detected: one player aged 9 and one aged 91, which warrant verification or careful handling

**Data Collection Method**

User registration/profile data
- signup forms
- Demographics entered by users 
- Self selected experience level or system assigned 

Session logging 
- System timestamps when users start/end activities 
- Linked via hashed email for privacy 

# (2) Question: